code/_onclick/hud/robot.dm

```diff
@@ -98,6 +98,13 @@
 	mymob.throw_icon.icon_state = "store"
 	mymob.throw_icon.name = "store"
 	mymob.throw_icon.screen_loc = ui_borg_store
+	
+//Headlamp
+	mymob:lamp_button = new /obj/screen()
```

Comment:
I wholeheartedly agree, but I think it's better if that's done in one sweep by updating our HUD code to /tg/'s latest.